src/common/types/types.cpp

```diff
@@ -11,48 +11,112 @@ namespace common {
 DataType::DataType() : typeID{ANY}, childType{nullptr} {}
 
 DataType::DataType(DataTypeID typeID) : typeID{typeID}, childType{nullptr} {
-    assert(typeID != LIST);
+    assert(typeID != VAR_LIST && typeID != FIXED_LIST);
 }
 
 DataType::DataType(DataTypeID typeID, std::unique_ptr<DataType> childType)
     : typeID{typeID}, childType{std::move(childType)} {
-    assert(typeID == LIST);
+    assert(typeID == VAR_LIST);
 }
 
-DataType::DataType(const DataType& other) : typeID{other.typeID} {
-    if (other.childType) {
+DataType::DataType(
+    DataTypeID typeID, std::unique_ptr<DataType> childType, uint64_t fixedNumElementsInList)
+    : typeID{typeID}, childType{std::move(childType)}, fixedNumElementsInList{
+                                                           fixedNumElementsInList} {
+    assert(typeID == FIXED_LIST);
+}
+
+DataType::DataType(const DataType& other) {
+    switch (other.typeID) {
+    case FIXED_LIST:
+    case VAR_LIST: {
+        typeID = other.typeID;
         childType = other.childType->copy();
+        fixedNumElementsInList = other.fixedNumElementsInList;
+    } break;
+    case ANY:
+    case NODE:
+    case REL:
+    case INTERNAL_ID:
+    case BOOL:
+    case INT64:
+    case DOUBLE:
+    case DATE:
+    case TIMESTAMP:
+    case INTERVAL:
+    case STRING: {
+        typeID = other.typeID;
+    } break;
+    default:
+        assert(false);
```

Comment:
throw InteralException instead of assert(). Assert will cause seg fault in release mode.

Reply:
changed to internalException